### ISSUE TITLE
Add some simple defence against spammers

### ISSUE THREAD

**Issue by rggibson** [rggibson/pb-tracker#45]
Just have a "can't submit 2 runs within X seconds" restriction for now.  The free quota on GAE could get exhausted if one user spams the site with a ton of runs.

**Comment by rggibson** [comment]
Limiting users for certain requests is also important to defend against brute force attacks on the /asup protocol, for example.

**Comment by albinohat** [comment]
Might I recommend, in addition to hard limits, adding an API key? It would allow you to identify the PBT account accessing the API. You can accept it as a URL param. See https://github.com/peppy/osu-api/wiki for an example.

**Comment by rggibson** [comment]
Yep, I agree.  Related issue is here: https://github.com/rggibson/pb-tracker/issues/44
